src/Sphere.jl

```diff
@@ -2,10 +2,36 @@ struct Sphere{T} <: Manifold where {T}
     shape::T
 end
 
-project_tangent!(S::Sphere,w,x,v) = (w .= w - dot(x,v)*x)
-distance(S::Sphere,x,y) = norm(x-y) # FIXME
-exp!(S::Sphere,y,x,v) = retract!(S,y,x,v) # FIXME
-log!(S::Sphere,v,x,y) = project_tangent!(S,v,x,y-x) # FIXME
+project_tangent!(S::Sphere, w, x, v) = (w .= v .- dot(x, v).*x)
+distance(S::Sphere, x, y) = acos(dot(x, y))
+
+function exp!(S::Sphere, y, x, v)
+    nv = norm(S, x, v)
+    if nv ≈ 0.0
+        y .= x
+    else
+        y .= cos(nv).*x .+ (sin(nv)/nv).*v
+    end
+    return y
+end
+
+function log!(S::Sphere, v, x, y)
+    θ = acos(dot(x, y))
+    if θ ≈ 0.0
+        zero_tangent_vector!(S, v, x)
```

Comment:
Not certain this is great for AD. Would be nice to use the first order Taylor expansion around `θ = 0.0` instead of `zero_tangent_vector!`.

Reply:
Actually `log` is not very well conditionned for `x` close to `y`, so that `approx` can be replaced with `==` and have the same behavior. There's no way to get an accurate result here without doing something like `log1p`.

Reply:
> Actually `log` is not very well conditionned for `x` close to `y`

That's interesting, can you explain a bit more?

Reply:
```
julia> rel_prec(x,y) = (x-y)/y
rel_prec (generic function with 1 method)

julia> rel_prec(log(1+1e-12),1e-12)
8.890058184100751e-5

julia> rel_prec(log1p(1e-12),1e-12)
-5.000242179397697e-13
```
Essentially, whenever `x` is close to `y`, you can't compute `x-y` to good relative accuracy.

Reply:
In particular here, isapprox(1-y,0.0) is zero if and only if `y==1`.